test/test_corner_detection.jl

```diff
@@ -0,0 +1,73 @@
+function _detect_corners(file, n_corners)
+    img = load(file)
+    gry = img[1:1, :, :]
+    cv_n_corners = OpenCV.Size{Int32}(n_corners...)
+    _cv_corners = OpenCV.Mat(Array{Float32}(undef, 2, 1, prod(n_corners)))
+    ret, cv_corners = OpenCV.findChessboardCorners(gry, cv_n_corners, _cv_corners, 0)
+    @assert ret "Failed to detect any corners!"
+    corners = reshape(vec.(eachslice(cv_corners, dims = 3)), n_corners)
+    return corners
+end
+
+function parse_corners_file(file)
+    open(file, "r") do o
+        readuntil(o, "rows:")
+        rows = parse(Int, readuntil(o, "\n"))
+        readuntil(o, "cols:")
+        cols = parse(Int, readuntil(o, "\n"))
+        readuntil(o, "[")
+        corners = [Float32[] for i in 1:cols, j in 1:rows]
+        for i in 1:cols*rows, j in 1:2
+            if i == cols*rows && j == 2
+                break
+            end
+            push!(corners[i], parse(Float32, readuntil(o, ",")))
+        end
+        push!(corners[cols*rows], parse(Float32, readuntil(o, "]")))
```

Comment:
If I understand correctly, this parses a matrix like [this](https://github.com/opencv/opencv_extra/blob/a74cf6bae7fd75d91282b877c559168b3a62148a/testdata/cv/cameracalibration/chess_corners1.dat#L8-L25) into a matrix-of-vectors where each vector holds a single value. I'm not familiar with OpenCV, so can you explain the reason for this choice? Why not `readdlm` the matrix directly?

Reply:
I think you can use `",]"` as the delimiter for `readuntil` and avoid the need for special-casing `i==cols*rows`.

Reply:
I had the same thought, it's simpler now (my only reason to do it this way was that it takes less memory, but hardly worth the effort). The real solution would have been if https://github.com/JuliaData/YAML.jl would have parsed that file, but it doesn't:
```julia
julia> using YAML

julia> data = YAML.load_file("chess_corners1.dat")
ERROR: expected '<document start>' but found YAML.DirectiveToken at nothing
Stacktrace:
  [1] parse_document_start(stream::YAML.EventStream)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/parser.jl:176
  [2] parse_implicit_document_start(stream::YAML.EventStream)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/parser.jl:153
  [3] peek(stream::YAML.EventStream)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/parser.jl:52
  [4] compose_document(composer::YAML.Composer)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/composer.jl:44
  [5] compose(events::YAML.EventStream, resolver::YAML.Resolver)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/composer.jl:33
  [6] load(tokenstream::YAML.TokenStream, constructor::YAML.Constructor)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:77
  [7] load
    @ ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:84 [inlined]
  [8] load(input::IOStream, more_constructors::Nothing, multi_constructors::Dict{Any, Any}; kwargs::@Kwargs{})
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:87
  [9] load (repeats 3 times)
    @ ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:87 [inlined]
 [10] #16
    @ ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:155 [inlined]
 [11] open(::YAML.var"#16#17"{@Kwargs{}, Tuple{}}, ::String, ::Vararg{String}; kwargs::@Kwargs{})
    @ Base ./io.jl:410
 [12] open
    @ ./io.jl:407 [inlined]
 [13] load_file(::String)
    @ YAML ~/.julia/packages/YAML/XZZ1Y/src/YAML.jl:153
 [14] top-level scope
    @ REPL[4]:1
```